xdsl/ir/core.py

```diff
@@ -708,6 +708,33 @@ def __hash__(self):
         return hash(self._op._operands)  # pyright: ignore[reportPrivateUsage]
 
 
+class OpTraits(Iterable[OpTrait]):
+    _traits: set[OpTrait] | Callable[[], set[OpTrait]]
+
+    def __init__(self, traits: set[OpTrait] | Callable[[], set[OpTrait]]) -> None:
+        self._traits = traits
+
+    def get_traits(self) -> set[OpTrait]:
```

Comment:
Wouldn't having `@property` and `def traits` being a bit more pythonic here?

Reply:
Yep, will do this now.

Reply:
I didn't want to before but I've changed my mind. I think it might make sense to make it a frozenset also, since mutation will be so infrequent.

Reply:
Done and done